src/dmd/target.d

```diff
@@ -774,6 +779,23 @@ struct Target
         const sz = t.size(loc);
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
+    /**
+     * Checks whether the target/compiler supports the given pragma.
+     * Params:
+     *  ident     = name of the pragma
+     *  sc        = scope of evaluation
+     *  args      = arguments to pass to pragma
+     *  ps        = if != null analysing a PragmaStatement
+     *  pd        = if != null analysing a PragmaDeclaration
+     * Returns:
+     *  true if the pragma is a known pragma
+     *  false otherwise.
+     */
+    extern(C++) static bool pragmaSemantic(Identifier ident, Scope* sc, Expressions* args,
+                                           PragmaStatement ps, PragmaDeclaration pd)
```

Comment:
Rather than passing both kinds of pragma to the function, just have a boolean true/false like I mentioned in the issue?

Reply:
You can't issue errors for misuse if you don't have the declaration/statement.

Reply:
Hmm, could we do without one or two parameters then? For instance args is part of the pragma declaration/statement itself, no need to pass it twice I guess.

Reply:
We could, I have it as a local of `PragmaSemanticAnalysis` because it is so widely used. Looking now I see that LDC does not support (non DMD) PragmaStatements, what is GDC's usage like?

Reply:
I have a use case that isn't implemented yet.
```
pragma(GNU_unroll, 20)
for (int i = 0; i < m; i++)
    a[i] = a[i + k] * c;
```